### ISSUE TITLE
Parameters are not used in propagated replaceable model with a constrainedby clause 

### ISSUE THREAD

**Issue by FWuellhorst** [OpenModelica/OpenModelica#9410]
### Description

If I propagate a replaceable model option and want to pass a parameter in the constrainedby clause, the parameter is not used.
This works as expected in Dymola. 
Following the MLS and this claytex post, I though this should be possible: 
https://modelica.org/documents/MLS.pdf
https://www.claytex.com/blog/replaceable-model-declaration-propagation/

### Steps to Reproduce

I wrote this small package. Run `Example` to reproduce:

```
package TestModelPropagation

partial model PartialSubsystem
  parameter Real p;
  Real y;
end PartialSubsystem;

model QuadraticP
  extends PartialSubsystem;
  
  equation
  y = p * p;
end QuadraticP;

model UppermostSystem
  parameter Real p2;
  replaceable model MySubsystem = PartialSubsystem constrainedby PartialSubsystem(p=p2);
  MySubsystem subsys;
  Real y;
  
  equation
    y = subsys.y;
  
end UppermostSystem;

model Example

  UppermostSystem sys(
    p2=100,
    redeclare model MySubsystem = QuadraticP);
end Example;
  
end TestModelPropagation;
```

### Expected Behavior

`sys.y` should equal 100^2 but is zero. Further, the parameter `p` should have the value 100.

### Version and OS

 - OpenModelica Version: 
Connected to OpenModelica v1.18.0 (64-bit)
Connected to OMSimulator v2.1.1.post80-g1bf17f4-mingw
 - OS: Windows 10, 64 bit
 - Versions of used Modelica libraries if applicable: MSL 4.0

### Additional Context

This issue prevents the simulation of a modular heat pump. Read this issue for further information: https://github.com/ibpsa/modelica-ibpsa/issues/1576

**Comment by perost** [comment]
I think this issue has already been fixed, so please try the 1.19.2 release instead and see if that works for you.

**Comment by FWuellhorst** [comment]
Thanks for the quick response! Yes, it is fixed! :)
